File: platformconnection/src/main/java/com/example/learningwebflux/platformconnection/ModuleIntegrationEventReceiversConfiguration.java
package com.example.learningwebflux.platformconnection;

import com.example.learningwebflux.common.integrationeventbus.ModuleIntegrationReceiversRegistry;
import com.example.learningwebflux.platformconnection.integrationeventreceivers.usersignedup.CreateCustomerIEReceiver;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class ModuleIntegrationEventReceiversConfiguration {
    @Bean
    ModuleIntegrationReceiversRegistry platformConnectionModuleIntegrationEventReceiversRegistry() {
        return new ModuleIntegrationReceiversRegistry(
            List.of(
                CreateCustomerIEReceiver.class
            )
        );
    }
}

File: notification/src/main/java/com/example/learningwebflux/notification/ModuleIntegrationEventReceiversConfiguration.kt
package com.example.learningwebflux.notification

import com.example.learningwebflux.common.integrationeventbus.ModuleIntegrationReceiversRegistry
import com.example.learningwebflux.notification.eventreceivers.userregistered.CreateUserOnSignedUpIE
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class ModuleIntegrationEventReceiversConfiguration {
    @Bean
    open fun notificationModuleIntegrationEventReceiversRegistry(): ModuleIntegrationReceiversRegistry {
        return ModuleIntegrationReceiversRegistry(
            listOf(
                CreateUserOnSignedUpIE::class.java,
            )
        )
    }
}